packages/cozy-client/src/registry.js

```diff
@@ -0,0 +1,55 @@
+import 'url-search-params-polyfill'
+import termUtils from './terms'
+import { APP_TYPE } from './constants'
+
+const queryPartFromOptions = options => {
+  const query = new URLSearchParams(options).toString()
+  return query ? `?${query}` : ''
+}
+
+class Registry {
+  constructor(options) {
+    if (!options.client) {
+      throw new Error('Need to pass a client to instantiate a Registry API.')
+    }
+    this.client = options.client
+  }
+  /**
+   * Installs or updates an app from a source.
+   *
+   * Accepts the terms if the app has them.
+
+   * @param  {RegistryApp} app - App to be installed 
+   * @param  {string} source - String (ex: registry://drive/stable)
+   * @return {Promise}
+   */
+  async installApp(app, source) {
+    const { slug, type, terms } = app
+    const searchParams = {}
+    const isUpdate = app.installed
+    if (isUpdate) searchParams.PermissionsAcked = isUpdate
+    if (source) searchParams.Source = source
+    const querypart = queryPartFromOptions(searchParams)
+    if (terms) {
```

Comment:
Shouldn't we call that explicitly in a distinct call ? I fear that we forget that we make our user accept implicitly and that we accept some terms in his name without him knowing.

Reply:
Seen together. The frontend code should propose the terms and permissions to the user but there is hardly a meaning to install an app without accepting the terms. Let's leave that as is